solution/parser/fr-parser/main.go

```diff
@@ -0,0 +1,183 @@
+	if os.Getenv("PARSER_ON_LAMBDA") == "true" {
+		lambda.Start(lambdaHandler)
+	} else if err := start(); err != nil {
+		log.Fatal("[main] Parser failed: ", err)
+	}
+}
+
+func getLogLevel(l string) log.Level {
+	switch l {
+	case "warn":
+		return log.WarnLevel
+	case "fatal":
+		return log.FatalLevel
+	case "error":
+		return log.ErrorLevel
+	case "info":
+		return log.InfoLevel
+	case "debug":
+		return log.DebugLevel
+	case "trace":
+		return log.TraceLevel
+	default:
+		log.Warn("[main] \"", l, "\" is an invalid log level, defaulting to \"warn\".")
+		return log.WarnLevel
+	}
+}
+
+var retrieveConfigFunc = ecfrEregs.RetrieveConfig
+var getLogLevelFunc = getLogLevel
+
+func loadConfig() (*ecfrEregs.ParserConfig, error) {
+	log.Info("[main] Loading configuration...")
+	config, _, err := retrieveConfigFunc()
+	if err != nil {
+		return nil, err
+	}
+
+	// parse config here
+	log.SetLevel(getLogLevel(config.LogLevel))
+
+	return config, nil
+}
+
+var extractSubchapterPartsFunc = ecfr.ExtractSubchapterParts
+
+func getPartsList(ctx context.Context, t *ecfrEregs.TitleConfig) []string {
+	today := time.Now()
+	var parts []string
+	for _, subchapter := range t.Subchapters {
+		subchapterParts, err := extractSubchapterPartsFunc(ctx, today, t.Title, &ecfr.SubchapterOption{subchapter[0], subchapter[1]})
+		if err != nil {
+			log.Error("[main] Failed to retrieve parts for title ", t.Title, " subchapter ", subchapter, ". Skipping.")
+			continue
+		}
+		parts = append(parts, subchapterParts...)
+	}
+	parts = append(parts, t.Parts...)
+	return parts
+}
+
+var loadConfigFunc = loadConfig
+var getPartsListFunc = getPartsList
+var processPartFunc = processPart
+
+func start() error {
+	ctx, cancel := context.WithTimeout(context.Background(), TIMELIMIT)
+	defer cancel()
+
+	config, err := loadConfigFunc()
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve configuration: %+v", err)
+	}
+
+	for _, title := range config.Titles {
+		log.Info("[main] Retrieving content for title ", title.Title)
+		parts := getPartsListFunc(ctx, title)
+		for _, part := range parts {
+			if err := processPartFunc(ctx, title.Title, part); err != nil {
+				log.Error("[main] Failed to process title ", title.Title, " part ", part, ": ", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+var fetchContentFunc = fedreg.FetchContent
+var processDocumentFunc = processDocument
+
+func processPart(ctx context.Context, title int, part string) error {
+	log.Debug("[main] Retrieving list of content for title ", title, " part ", part)
+	content, err := fetchContentFunc(ctx, title, part)
+	if err != nil {
+		return fmt.Errorf("Fetch content failed: %+v", err)
+	}
+
+	log.Debug("[main] Processing content for title ", title, " part ", part)
+	for _, c := range content {
+		if err := processDocumentFunc(ctx, title, part, c); err != nil {
+			log.Error("[main] Failed to process title ", title, " part ", part, " doc ID ", c.DocumentNumber, ": ", err)
```

Comment:
If this isn't raised as an error, should it be log.Warn?

Reply:
Only log.Fatal changes the return code. I see Warn as "something bad happened but it's not a big deal" and Error is "something failed but I can continue anyway".